src/app/legacy/containers/App/App.jsx

```diff
@@ -29,83 +26,36 @@ const mapToState = ({ pathname, initialData, routeProps, toggles }) => {
   ]);
 };
 
-const getNextPageState = async pathname => {
-  const routeProps = getRouteProps(pathname);
-  const { service, variant, route, isAmp } = routeProps;
-  const { pageType, getInitialData } = route;
-  const toggles = await getToggles(service);
-  const initialData = await getInitialData({
+export const App = ({ initialData, bbcOrigin }) => {
+  const {
     path: pathname,
-    service,
-    variant,
-    pageType,
+    showAdsBasedOnLocation,
     toggles,
-    isAmp,
-  });
-
-  return mapToState({ pathname, initialData, routeProps, toggles });
-};
-
-const setFocusOnMainHeading = () => {
-  const mainHeadingEl = document.querySelector('h1#content');
-
-  if (mainHeadingEl) {
-    mainHeadingEl.focus();
-  }
-};
+    mvtExperiments,
+    isUK,
+  } = initialData;
 
-export const App = ({ location, initialData, bbcOrigin, history }) => {
-  const { pathname } = location;
-  const hasMounted = useRef(false);
   const routeProps = getRouteProps(pathname);
-  const previousPath = useRef(null);
-  const { showAdsBasedOnLocation, toggles, mvtExperiments, isUK } = initialData;
-  const [state, setState] = useState(
+
+  const [state] = useState(
     mapToState({
       pathname,
       initialData,
       routeProps,
       toggles,
     }),
   );
-  const routeHasChanged = state.pathname !== pathname;
-
-  useEffect(() => {
-    if (history.action === 'POP') {
-      previousPath.current = null; // clear the previous path on back clicks
-    }
-    return () => {
-      previousPath.current = pathname;
-    };
-  }, [pathname, history]);
 
-  useEffect(() => {
-    if (hasMounted.current) {
-      getNextPageState(pathname).then(setState);
-    } else {
-      hasMounted.current = true;
-    }
-  }, [pathname]);
-
-  useLayoutEffect(() => {
-    if (hasMounted.current) {
-      if (routeHasChanged) {
-        window.scrollTo(0, 0);
-      } else {
-        setFocusOnMainHeading();
-      }
-    }
-  }, [routeHasChanged]);
+  const routeHasChanged = state.pathname !== pathname;
```

Comment:
Do we need this anymore - seems related to `loading` below - can that also be removed?

Reply:
Some tests were checking the value, but other than that it did appear to be related to the withLoading container, so I think it's safe to be deleted.